app/javascript/components/async-credentials/check-errors.jsx

```diff
@@ -1,32 +1,30 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import { useFieldApi } from '@@ddf';
+
 const CheckErrors = ({
   names,
   subscription,
   fieldsState,
   children,
   originalRender,
-  FieldProvider,
   valid,
 }) => {
   if (!names.length) {
     return (originalRender || children)(valid);
   }
   const [name, ...rest] = names;
+  const fieldState = useFieldApi({ name, subscription });
+
   return (
-    <FieldProvider name={name} subscription={subscription}>
-      {fieldState => (
-        <CheckErrors
-          names={rest}
-          subscription={subscription}
-          originalRender={originalRender || children}
-          fieldsState={{ ...fieldsState, [name]: fieldState }}
-          FieldProvider={FieldProvider}
-          valid={[...valid, fieldState.meta.valid]}
-        />
-      )}
-    </FieldProvider>
+    <CheckErrors
```

Comment:
Should we just iterate over `names` now, instead of recursively rendering ourselves for each name?

Reply:
This component is no longer used, I do the iteration recursively one level above using `snapshot()`.